sqlglot/optimizer/merge_subqueries.py

```diff
@@ -195,7 +196,7 @@ def _is_recursive():
         and not outer_scope.expression.is_star
         and isinstance(inner_select, exp.Select)
         and not any(inner_select.args.get(arg) for arg in UNMERGABLE_ARGS)
-        and inner_select.args.get("from")
+        and inner_select.args.get("from") is not None
```

Comment:
What triggered this change?

Reply:
Iirc the first variant bubbles `t.Any` up the conjuction chain, so mypy complains that the return type is `t.Any | bool`, with this change it's silenced

Reply:
Ah I see, sounds good.